MFVNeutralino/plugins/TriggerFloats.cc

```diff
@@ -90,12 +103,55 @@ void MFVTriggerFloats::produce(edm::Event& event, const edm::EventSetup& setup)
     edm::Handle<l1t::JetBxCollection> l1_jets;
     event.getByToken(l1_jets_token, l1_jets);
 
+    edm::Handle<pat::METCollection> mets;
+    event.getByToken(met_token, mets);
+
     edm::Handle<pat::MuonCollection> muons;
     event.getByToken(muons_token, muons);
 
     edm::Handle<reco::VertexCollection> primary_vertices;
     event.getByToken(primary_vertex_token, primary_vertices);
 
+    const pat::MET& met = mets->at(0);
+    double met_pt_origin = met.pt();
+    double met_phi_origin = met.phi();
+    std::pair<double,double> met_corrected = jmt::METXYCorr_Met_MetPhi(met_pt_origin, met_phi_origin, event.id().event(), year, isMC, primary_vertices->size()); // year and isMC need to be modified
+    double met_pt = met_corrected.first;
+    double met_phi = met_corrected.second;
+    double met_px = met_pt*std::cos(met_phi);
+    double met_py = met_pt*std::sin(met_phi);
+    for (const pat::Muon& muon : *muons) {
+      double muon_px = muon.px();
+      double muon_py = muon.py();
+      //if (muon_selector(muon)){
+      if (muon.passed(reco::Muon::CutBasedIdTight)) {
+        met_px = met_px + muon_px;
```

Comment:
I think we should use the selector rather than hardcoding it here. Especially since we were using Loose elsewhere in the code! Maybe we aren't making extensive use of the MetNoMu right now, but I don't want us to get burned by something like this....